### PR TITLE
egressIP test suite added to the rds system test

### PR DESCRIPTION
**Pull request by elenagerman**
egressIP test suite added to the rds system test

### CODE REVIEW

File: tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go [hunk 1]
```diff
@@ -81,3 +91,280 @@ func VerifyOperatorDeployment(apiClient *clients.Settings,
...
+	deploymentName, nsName, deploymentLabel string) error {
+	glog.V(100).Infof("Removing test deployment %q from %q ns", deploymentName, nsName)
+
+	if snifferDeployment, err := deployment.Pull(apiClient, deploymentName, nsName); err == nil {
+		glog.V(100).Infof("Deleting deployment %q from %q namespace", deploymentName, nsName)
+
+		err = snifferDeployment.DeleteAndWait(300 * time.Second)
+
+		if err != nil {
+			return fmt.Errorf("failed to delete deployment %q from %q namespace: %w", deploymentName, nsName, err)
```
Comment (yprokule) [inline tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go:107]: nit: worth logging that error's happened during deletion as well as a successful deletion

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+		packetCaptureProtocol,
+		packetCaptureInterface,
+		tcpdumpImage,
+		snifferDeploymentName,
+		snifferNamespace,
+		scheduleOnNodes,
+	)
+
+	if err != nil {
+		return snifferDeployment, err
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:78]: worth logging an error

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+
+	containerCmd := []string{
+		"/bin/bash",
+		"-c",
+		fmt.Sprintf(cmd, packetCaptureInterface, packetCapturePort),
+	}
+
+	var err error
+
+	glog.V(100).Infof("Checking sniffer deployment don't exist")
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:164]: nit: `doesn't`

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+	}
+
+	matchesFound := 0
+
+	var snifferPodObj *pod.Builder
+
+	for _, podObj := range snifferPodObjects {
+		if podObj.Object.Spec.NodeName == proberPod.Object.Spec.NodeName {
+			glog.V(100).Infof("sniffer pod: %s", podObj.Definition.Name)
+			snifferPodObj = podObj
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:354]: shouldn't the loop be interrupted when the 1st matching pod is found?

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+
+	if len(logs) != 0 {
+		buf := new(bytes.Buffer)
+		_, err = buf.WriteString(logs)
+
+		if err != nil {
+			return 0, fmt.Errorf("error in copying info from pod logs to buffer")
+		}
+
+		_ = buf.String()
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:392]: why is this needed?

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 6]
```diff
@@ -0,0 +1,633 @@
...
+	if proberPod == nil {
+		return "", fmt.Errorf("nil pointer to proberPod was provided in sendProbesToHostPort")
+	}
+
+	if targetProtocol != httpProtocol && targetProtocol != udpProtocol {
+		return "", fmt.Errorf("sendProbesToHostPort supports only %s and %s protocols",
+			httpProtocol, udpProtocol)
+	}
+
+	targetURL := fmt.Sprintf("%s:%d", targetHost, targetPort)
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:432]: U might need extra check for the IPv6 address as it should be wrapped with square brackets

File: tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go [hunk 7]
```diff
@@ -81,3 +91,280 @@ func VerifyOperatorDeployment(apiClient *clients.Settings,
...
+		}
+	} else {
+		glog.V(100).Infof("deployment %q not found in %q namespace", deploymentName, nsName)
+	}
```
[the remark this answers is not in the document]
Comment (elenagerman) [reply tests/system-tests/internal/apiobjectshelper/apiobjectshelper.go:111]: I have nothing to do with it since I do not have any idea how to differentiate such error.
I do not see any of implemented by you Delete... methods includes such verification and I never meet such failure before

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 8]
```diff
@@ -0,0 +1,633 @@
...
+
+	glog.V(100).Infof("Creating the sniffer deployment with image %s", networkPacketSnifferImage)
+
+	cmd := tcpCaptureScript
+	if packetCaptureProtocol == udpProtocol {
+		cmd = udpCaptureScript
+	}
+
+	if strings.Contains(packetCaptureInterface, ".") {
+		packetCaptureInterface = strings.Split(packetCaptureInterface, ".")[0]
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:153]: in case of a vlan sub interface - why to listen on the _parent_ and not on the actual interface?

File: tests/system-tests/internal/sniffer/sniffer.go [hunk 9]
```diff
@@ -0,0 +1,633 @@
...
+const (
+	httpProtocol           = "http"
+	udpProtocol            = "udp"
+	snifferDeployRBACName  = "privileged-rdscore-sniffer"
+	snifferDeploySAName    = "default"
+	snifferRBACRole        = "system:openshift:scc:privileged"
+	snifferDeploymentLabel = "rds-core=sniffer-deploy"
+
+	tcpCaptureScript = `#!/bin/bash
+tcpdump -s 0 -A -v -i %s | egrep -i 'Host:'
```
Comment (yprokule) [inline tests/system-tests/internal/sniffer/sniffer.go:33]: why not to narrow requests with the destination?
```suggestion
tcpdump -s 0 -A -v -i %s host %s | egrep -i 'Host:'
```
